modules/web/js/ballerina/diagram2/views/default/components/nodes/struct-node.jsx

```diff
@@ -187,6 +188,67 @@ class StructNode extends React.Component {
     }
 
     /**
+     * Open JSON import dialog box and pass callback
+     */
+    onClickJsonImport() {
+        const onImport = (json) => {
+            let refExpr = TreeBuilder.build(FragmentUtils.parseFragment(FragmentUtils.createExpressionFragment(json)));
+            let currentValue;
+            this.props.model.setFields([], true);
+            refExpr.variable.initialExpression.keyValuePairs.forEach((ketValPair) => {
```

Comment:
Also do a isRefExpr check if we are not validating the json content to be json in the text area.

Reply:
JSON will be validated from the ACE editor. Import button will be only enabled if its a valid JSON